src/main/java/me/zeroeightsix/kami/module/modules/movement/AutoWalk.java

```diff
@@ -33,25 +34,63 @@
             case BACKWARDS:
                 event.getMovementInput().moveForward = -1;
                 break;
-            case PATH:
-                if (Pathfind.points.isEmpty()) return;
-                event.getMovementInput().moveForward = 1;
-                if (mc.player.isInWater() || mc.player.isInLava()) mc.player.movementInput.jump = true;
-                else if (mc.player.collidedHorizontally && mc.player.onGround) mc.player.jump();
-                if (!MODULE_MANAGER.isModuleEnabled(Pathfind.class) || Pathfind.points.isEmpty()) return;
-                PathPoint next = Pathfind.points.get(0);
-                lookAt(next);
-                break;
         }
     });
 
-    private void lookAt(PathPoint pathPoint) {
-        double[] v = calculateLookAt(pathPoint.x + .5f, pathPoint.y, pathPoint.z + .5f, mc.player);
-        mc.player.rotationYaw = (float) v[0];
-        mc.player.rotationPitch = (float) v[1];
+    @EventHandler
+    public Listener<ServerDisconnectedEvent> kickListener = new Listener<>(event -> {
+        if(mode.getValue().equals(AutoWalkMode.BARITONE) && isEnabled()) {
+            disable();
+        }
+    });
+
+    public void onEnable() {
```

Comment:
This would look MUCH nicer written in Kotlin, and it's also the textbook example of when you should use a switch

Reply:
ik it looks like yandev code, I tried that as well, but switches are like nah chief when it comes to booleans.

Reply:
I'll fix that then.